mlrun/projects/pipelines.py

```diff
@@ -354,11 +358,18 @@ def enrich_function_object(
         f.enrich_runtime_spec(
             project.spec.default_function_node_selector,
         )
-
+    skip_automount_for_kfp = False
+    if project.spec.workflows:
+        skip_automount_for_kfp = (
+            project.spec.workflows[0].get("engine")
+            == mlrun_pipelines.common.models.EngineType.KFP
+        )
```

Comment:
Saar means that you are checking if the project HAS a workflow but this may disable auto mount for regular jobs as well since they may pass here so the check is no accurate.
Project object having a workflow spec doesn't mean that this workflow is what's currently running.

Reply:
discussed offline, use `pipeline_context` here